src/CommandingOfficers/DefendPeace/RoseThorn/Tech.java

```diff
@@ -570,23 +570,27 @@ private static UnitModel createBattleMechModel(GameReadyModels grms)
     UnitModel mdTank = grms.unitModels.getAllModels(UnitModel.LAND | UnitModel.ASSAULT, matchOnAny).get(1);
     UnitModel antiAir = grms.unitModels.getUnitModel(UnitModel.LAND | UnitModel.SURFACE_TO_AIR, matchOnAny);
 
-    UnitModel BattleMech = mdTank.clone();
-    BattleMech.name = BATTLEMECH_NAME;
-    BattleMech.role = BattleMech.role | UnitModel.SURFACE_TO_AIR;
-    BattleMech.costBase = mdTank.costBase*2 + (antiAir.costBase/2);
-    BattleMech.abilityPowerValue = 20;
-    BattleMech.maxFuel = 30;
-    BattleMech.maxAmmo = 10;
-    BattleMech.visionRange = 2;
-    BattleMech.baseMovePower = 4;
-    BattleMech.baseMoveType = new FootMech();
-    BattleMech.healableHabs = new HashSet<TerrainType>(); // BattleMechs have specialized parts, not easy to repair.
+    var BattleMech = mdTank.toBuilder();
+    BattleMech.name(BATTLEMECH_NAME);
+    BattleMech.costBase(mdTank.costBase*2 + (antiAir.costBase/2));
+    BattleMech.abilityPowerValue(20);
+    BattleMech.maxFuel(30);
+    BattleMech.maxAmmo(10);
+    BattleMech.visionRange(2);
+    BattleMech.baseMovePower(4);
+    BattleMech.baseMoveType(new FootMech());
 
-    WeaponModel ratatat = antiAir.weapons.get(0).clone();
     ArrayList<WeaponModel> weapons = new ArrayList<WeaponModel>();
-    weapons.add(BattleMech.weapons.get(0));
+    weapons.add(mdTank.weapons.get(0));
+    WeaponModel ratatat = antiAir.weapons.get(0).clone();
```

Comment:
Why do we clone the AA, but not the Md Tank weapon? I know it was like that before, but still. Past me should have left a comment, probs.

Reply:
I think we were cloning as a matter of general principles before, but I don't think there's much point now.